pkg/schedule/schedulers/scheduler_controller.go

```diff
@@ -41,6 +41,7 @@ var denySchedulersByLabelerCounter = labeler.LabelerEventCounter.WithLabelValues
 type Controller struct {
 	sync.RWMutex
 	wg      sync.WaitGroup
+	wgLock  sync.Mutex
```

Comment:
I think the problem is not whether we use mutex to protect it.

Reply:
Do you have any other ideas?